packages/block-library/src/playlist/index.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Server-side rendering of the `core/playlist` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/playlist` block on server.
+ *
+ * @since 6.8.0
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the Playlist.
+ */
+function render_block_core_playlist( $attributes, $content ) {
+	if ( empty( $attributes['tracks'] ) ) {
+		return '';
+	}
+
+	$current_media_id = $attributes['currentTrack'];
+
+	/**
+	 * Returns early if no valid track ID is found.
+	 * This can happen if the user deleted all tracks but kept an empty inner
+	 * block, such as the media upload placeholder.
+	 */
+	if ( empty( $current_media_id ) ) {
+		return '';
+	}
+
+	wp_enqueue_script_module( '@wordpress/block-library/playlist/view' );
+
+	wp_interactivity_state(
+		'core/playlist',
+		array(
+			'currentTrack'   => function () {
+				$state = wp_interactivity_state();
+				$context = wp_interactivity_get_context();
+				return $state['tracks'][ $context['currentId'] ];
+			},
+			'isCurrentTrack' => function () {
+				$context = wp_interactivity_get_context();
+				return $context['currentId'] === $context['id'];
+			},
+		)
+	);
+
+	// Finds the unique id of the current track and populates the playlist array.
+	$p               = new WP_HTML_Tag_Processor( $content );
+	$playlist_tracks = array();
+	while ( $p->next_tag( 'button' ) ) {
+		$track_context     = $p->get_attribute( 'data-wp-context' );
+		$track_unique_id   = json_decode( $track_context, true )['id'];
+		$state             = wp_interactivity_state( 'core/playlist' );
+		$playlist_tracks[] = $track_unique_id;
+		if (
+			isset( $state['tracks'][ $track_unique_id ]['media_id'] ) &&
+			$state['tracks'][ $track_unique_id ]['media_id'] === $current_media_id
+		) {
+			$current_unique_id = $track_unique_id;
+		}
+	}
```

Comment:
I'm using your pattern to populate the array with all the tracks and the current track.

I haven't put any guard against undefined/null values, so it would be great if you could take a look. Thanks!